packages/backend/src/apps/gathersg/dynamic-data/get-case-fields.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  DynamicDataOutput,
+  IDynamicData,
+  IGlobalVariable,
+} from '@plumber/types'
+
+import HttpError from '@/errors/http'
+import computeParameters from '@/helpers/compute-parameters'
+import ExecutionStep from '@/models/execution-step'
+
+import { GatherSGError } from '../common/types'
+
+const dynamicData: IDynamicData = {
+  key: 'getCaseFields',
+  name: 'Get Case Fields',
+  async run($: IGlobalVariable): Promise<DynamicDataOutput> {
+    try {
+      // This action only allows a step variable which we have to attempt to compute the parameter value, thinking if there is a better way to do this
+      // TODO: see if we can refresh the case fields from the API instead of using the cached data because right now, the user has to manually refresh the case fields to get the latest data
+      const { caseId } = $.step.parameters
+      if (!caseId) {
+        return {
+          data: [],
+        }
+      }
+
+      const priorExecutionSteps = await ExecutionStep.query().where({
+        execution_id: $.flow.testExecutionId,
+        status: 'success',
+      })
+
+      const computedParameters = computeParameters(
+        $.step.parameters,
+        priorExecutionSteps,
+      )
+      const computedCaseId = computedParameters.caseId as string
+
+      const { data: responseData } = await $.http.get(
+        `/cases/${computedCaseId}`,
```

Comment:
let's try to use urlPathParams as much as possible lol

e.g.
export async function getTemplateData($: IGlobalVariable) {
  return await $.http.get('/v1/templates/:templateId', {
    urlPathParams: {
      templateId: $.step.parameters.templateId,
    },
  })
}

Reply:
thks for the catch naise